### PR TITLE
Add C-wrapper and demo

### PR DESCRIPTION
**Pull request by dirkschulze**
#16

### CODE REVIEW

**Comment by dirkschulze** [comment]
@mpsuzuki I think this ready for review now.

File: svgnative/CMakeLists.txt [hunk 1]
```diff
@@ -1,396 +1,420 @@
...
+CMAKE_DEPENDENT_OPTION(USE_SHARED "Dynamic library" TRUE "SHARED" FALSE)
+
+CMAKE_DEPENDENT_OPTION(USE_TEXT "Text port for testing" TRUE "TEXT" FALSE)
+CMAKE_DEPENDENT_OPTION(USE_CG "CoreGraphics needs to be enabled and run on macOS/iOS" TRUE "APPLE;CG" FALSE)
+CMAKE_DEPENDENT_OPTION(USE_SKIA "Skia" TRUE "APPLE;SKIA" FALSE)
+CMAKE_DEPENDENT_OPTION(USE_GDIPLUS "GDI+ port" TRUE "MSVC;GDIPLUS" FALSE)
+CMAKE_DEPENDENT_OPTION(USE_CAIRO "Cairo" TRUE "CAIRO" FALSE)
+
+CMAKE_DEPENDENT_OPTION(USE_TEXT_EXAMPLE "Text port for testing" TRUE "NOT LIB_ONLY;TEXT" FALSE)
+CMAKE_DEPENDENT_OPTION(USE_C_EXAMPLE "C Wrapper port for demonstration" TRUE "NOT LIB_ONLY;TEXT" FALSE)
```
Comment (dirkschulze) [inline svgnative/CMakeLists.txt]: This was added.

File: svgnative/CMakeLists.txt [hunk 2]
```diff
@@ -1,396 +1,420 @@
...
+  add_definitions(-DVER_PRODUCTVERSION=${VER_FILEVERSION})
+  add_definitions(-DVER_PRODUCTVERSION_STR="${VERSION}")
+
+endif()
+
+if(STYLE)
+    # Enable CSS style support. This feature is deprecated and will get removed.
+    add_definitions(-DSTYLE_SUPPORT)
+endif()
+# Expose CMake definitions as preprocessor macros
```
Comment (dirkschulze) [inline svgnative/CMakeLists.txt]: The following lines were added.

File: svgnative/CMakeLists.txt [hunk 3]
```diff
@@ -1,396 +1,420 @@
...
+add_subdirectory(example/testC)
+endif()
+if (USE_CAIRO_EXAMPLE)
+add_subdirectory(example/testCairo)
+endif()
+
+##############################
+# CPP and Headers for Library
+##############################
+file(GLOB gl_public
```
Comment (dirkschulze) [inline svgnative/CMakeLists.txt]: SVGNativeWrapper.h/.cpp were added.

File: svgnative/CMakeLists.txt [hunk 4]
```diff
@@ -1,396 +1,420 @@
...
+##############################
+# Library search paths
+##############################
+set(PRIVATE_LIBS)
+set(REQUIRES)
+set(PRIVATE_REQUIRES)
+set(PORTS_INCLUDES)
+
+if(USE_CG)
+    target_link_libraries(SVGNativeViewerLib "-framework CoreGraphics")
```
Comment (dirkschulze) [inline svgnative/CMakeLists.txt]: @mpsuzuki Added CF framework.


Not sure why this file shows up so messed up. Commented on all actual changes of this file.

**Comment by dirkschulze** [comment]
@moazin @lemzwerg I'd like to hear your input to the C wrapper too.

**Comment by mpsuzuki** [comment]
Oh, I found svgnative/CMakeLists.txt uses CRLF (like MSDOS) instead of LF (like macOSX or Unix). All other files use LF.

**Comment by mpsuzuki** [comment]
Now I'm trying to test Skia on GNU/Linux, and found that SVGNativeCWrapper.cpp should include "SkImage.h" and "SkCanvas.h" to be compiled.

**Comment by dirkschulze** [comment]
Will add them. @mpsuzuki btw. Could you post your findings with regards to building Skia latest? Maybe we can add a README to the Skia port folder.

**Comment by mpsuzuki** [comment]
Yes, of course. One of my long running task is that enabling Skia port on Windows and GNU/Linux. One of my concern is what is the standard directory structure after the manual building of Skia.

If I understand the document correctly, the resulted libraries would be placed under skia/out/Static or skia/out/Shared.

Unfortunately, I cannot find the sustainable user community using Skia (as self-standing library) on Windows, and still wondering what directory structure is standard. The vcpkg developers are working to have a rebuildable source package of Skia (and Google Chromium), but not finished yet (due to stupid design of meta-building systems).

Is it acceptable to introduce some commandline options to specify the (non-standard) location of Skia libraries?

**Pull request merged**